File: task_equivalents/llm_translations/gpt/base/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(readr)
library(dplyr)
library(tibble)
library(caTools)

College <- read_csv("data/csv/College.csv")

school_name <- College[[1]]
College <- College %>%
column_to_rownames(var = names(College)[1])

College <- College %>%
mutate(Private01 = ifelse(Private == "Yes", 1, 0)) %>%
select(-Private)

X <- College %>%
select(-Apps) %>%
mutate(across(everything(), as.numeric))

Y <- College$Apps

set.seed(1)
split <- sample.split(Y, SplitRatio = 0.5)

xtrain <- X[split, ]
xtest <- X[!split, ]
ytrain <- Y[split]
ytest <- Y[!split]

File: task_equivalents/llm_translations/gpt/base/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import numpy as np
import pandas as pd
from patsy import dmatrices

College = pd.read_csv("data/csv/College.csv", header=0)
College.index = College.iloc[:, 0]
College = College.iloc[:, 1:]

np.random.seed(1)
train = np.random.choice(College.index, size=len(College) // 2, replace=False)
test = College.index.difference(train)

df_train = College.loc[train, :]
df_test = College.loc[test, :]

y_train, x_train = dmatrices("Apps ~ .", data=df_train, return_type="dataframe")
y_train = y_train.iloc[:, 0]
x_train = x_train.iloc[:, 1:]

y_test, x_test = dmatrices("Apps ~ .", data=df_test, return_type="dataframe")
y_test = y_test.iloc[:, 0]
x_test = x_test.iloc[:, 1:]